### ISSUE TITLE
Can't make deployed infantries turn

### ISSUE THREAD

**Issue by forcecore** [OpenRA/OpenRA#12738]
Problem arises when trying to implemented RA2 style deploy and attack infantry.

While it is possible to make units fire alternative weapon when deployed,
the problem is they are not turreted: they have to turn to attack. With Mobile trait disabled when deployed, they can't turn! Alternative method would be to have a dummy turret which look like the deployed infantry.

Well, that happens in TS mod so I looked at tick tank.
Unfortunately, deployed infantries don't have WithVoxelTurrents, as they are sprites.
For tick tanks,  WithVoxelTurret plays well with AttackTurreted and AttackFrontal traits.
However, for infantries, WithSpriteTurret crashes the game horribly when I build the infantry:

```
		public WithSpriteTurret(Actor self, WithSpriteTurretInfo info)
			: base(info)
		{
			rs = self.Trait<RenderSprites>();
			body = self.Trait<BodyOrientation>();
			Attack = self.TraitOrDefault<AttackBase>();
			t = self.TraitsImplementing<Turreted>()
				.First(tt => tt.Name == info.Turret);
			arms = self.TraitsImplementing<Armament>()
				.Where(w => w.Info.Tu
```
Because both AttackTurreted and AttackFrontal provide AttackBase.

Actor fant has multiple traits of type `OpenRA.Mods.Common.Traits.AttackBase`

```
Over Powered Mod Mod at Version release-20170102
on map 9a0be9fc201c0fe99407e2a2962b66629fb7b2cc (Tournament Island by Scott_NZ).
Date: 2017-02-10 03:42:36Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.InvalidOperationException`: Actor fant has multiple traits of type `OpenRA.Mods.Common.Traits.AttackBase`
   위치: OpenRA.TraitDictionary.TraitContainer`1.GetOrDefault(UInt32 actor) 파일 D:\work\OpenRA\OpenRA.Game\TraitDictionary.cs:줄 162
   위치: OpenRA.TraitDictionary.GetOrDefault[T](Actor actor) 파일 D:\work\OpenRA\OpenRA.Game\TraitDictionary.cs:줄 95
   위치: OpenRA.Actor.TraitOrDefault[T]() 파일 D:\work\OpenRA\OpenRA.Game\Actor.cs:줄 252
   위치: OpenRA.Mods.Common.Traits.Render.WithSpriteTurret..ctor(Actor self, WithSpriteTurretInfo info)
   위치: OpenRA.Mods.Common.Traits.Render.WithSpriteTurretInfo.Create(ActorInitializer init)
   위치: OpenRA.Actor..ctor(World world, String name, TypeDictionary initDict) 파일 D:\work\OpenRA\OpenRA.Game\Actor.cs:줄 102
   위치: OpenRA.World.CreateActor(Boolean addToWorld, String name, TypeDictionary initDict) 파일 D:\work\OpenRA\OpenRA.Game\World.cs:줄 250
   위치: OpenRA.World.CreateActor(String name, TypeDictionary initDict) 파일 D:\work\OpenRA\OpenRA.Game\World.cs:줄 245
   위치: OpenRA.Mods.Common.Traits.Production.<>c__DisplayClass9_0.<DoProduction>b__0(World w)
   위치: OpenRA.World.Tick() 파일 D:\work\OpenRA\OpenRA.Game\World.cs:줄 347
   위치: OpenRA.Game.InnerLogicTick(OrderManager orderManager) 파일 D:\work\OpenRA\OpenRA.Game\Game.cs:줄 560
   위치: OpenRA.Game.LogicTick() 파일 D:\work\OpenRA\OpenRA.Game\Game.cs:줄 585
   위치: OpenRA.Game.Loop() 파일 D:\work\OpenRA\OpenRA.Game\Game.cs:줄 712
   위치: OpenRA.Game.Run() 파일 D:\work\OpenRA\OpenRA.Game\Game.cs:줄 752
   위치: OpenRA.Program.Run(String[] args) 파일 D:\work\OpenRA\OpenRA.Game\Support\Program.cs:줄 121
   위치: OpenRA.Program.Main(String[] args) 파일 D:\work\OpenRA\OpenRA.Game\Support\Program.cs:줄 41
```

An alternative approach might be having two Mobile traits, where the other mobile trait has speed=0. That doesn't work as speed can't be 0. (induces div by 0 error).

Another way is to have the second mobile trait with really really slow speed but it doesn't work either since OpenRA engine can't cope with two Mobile traits. (Sounds like another problem as modders can't have units that speed boost as in RA3 Kirovs)

**Comment by forcecore** [comment]
Deploy animation not playing issue might be related to this issue.
https://github.com/OpenRA/OpenRA/issues/12632

It seems WithSpriteTurret trait uses AttackBase only to lookup animation sequence.

Both the issues happen because animation sequence provider is not unified.

**Comment by forcecore** [comment]
Maybe not, Attack state determination should be read from one of AttackTurreted, as the attack is coming from the turret? Then why not use 
			Attack = self.TraitOrDefault<AttackTurreted>();
 instead?

**Comment by forcecore** [comment]
The above change made the turreted approach work but now the problem is that the deployed infantry tries to face "neutral facing" when the combat is over.

**Comment by pchote** [comment]
The issue is that `WithSpriteTurret` does not support multiple `AttackBase` traits.  Compare this with #12650 and its fix.

**Comment by GraionDilach** [comment]
#11600 could have fixed this (and some other bugs) but then you started dismissing it due to https://github.com/OpenRA/OpenRA/pull/11600#discussion-diff-77333581 and then blamed me - http://logs.openra.net/?year=2016&month=09&day=09#19:53:22 - for not adhering to the false alarm.

**Comment by pchote** [comment]
This was fixed by #14846.
